src/toolkits/drawing_classifier/dc_data_iterator.cpp

```diff
@@ -164,8 +179,11 @@ data_iterator::batch simple_data_iterator::next_batch(size_t batch_size) {
     }
   }
 
+  DASSERT_EQ(batch_size, batch_targets.size());
+
   // Wrap the buffers as float_array values.
   data_iterator::batch result;
+  result.num_samples = batch_size;
```

Comment:
Will we always have `batch_size` samples? I was under the impression that DC trains by epoch... What do we do at the end of one epoch if the batch size does not evenly divide the number of training examples?

Reply:
Fixed!